UpdateService/handler.ts

```diff
@@ -0,0 +1,199 @@
+  context: Context,
+  auth: IAzureApiAuthorization,
+  clientIp: ClientIp,
+  attrs: IAzureUserAttributes,
+  serviceId: NonEmptyString,
+  servicePayload: ServicePayload
+) => Promise<ResponseTypes>;
+
+const getSubscriptionKeysTask = (
+  apiClient: ReturnType<APIClient>,
+  serviceId: NonEmptyString
+): TaskEither<ErrorResponses, SubscriptionKeys> =>
+  tryCatch(
+    () =>
+      apiClient.getSubscriptionKeys({
+        service_id: serviceId
+      }),
+    errs => ResponseErrorInternal(JSON.stringify(errs))
+  ).foldTaskEither(
+    err => fromLeft(err),
+    maybeResponse =>
+      maybeResponse.fold(
+        errs => fromLeft(ResponseErrorInternal(JSON.stringify(errs))),
+        responseType =>
+          responseType.status !== 200
+            ? fromLeft(toErrorServerResponse(responseType))
+            : taskEither.of(responseType.value)
+      )
+  );
+
+const updateServiceTask = (
+  apiClient: ReturnType<APIClient>,
+  servicePayload: ServicePayload,
+  serviceId: NonEmptyString,
+  subscriptionId: NonEmptyString
+): TaskEither<ErrorResponses, Service> =>
+  fromEither<IResponseErrorUnauthorized, {}>(
+    serviceId !== subscriptionId
+      ? left(
+          ResponseErrorUnauthorized(
+            "Unauthorized",
+            "You are not allowed to update this service"
+          )
+        )
+      : right({})
+  ).foldTaskEither(
+    e => fromLeft(e),
+    _ =>
+      tryCatch(
+        () =>
+          apiClient.updateService({
+            service: {
+              ...servicePayload,
+              authorized_recipients: [],
+              service_id: subscriptionId
+            },
+            service_id: serviceId
+          }),
+        errs => ResponseErrorInternal(JSON.stringify(errs))
+      ).foldTaskEither(
+        err => fromLeft(err),
+        maybeResponse =>
+          maybeResponse.fold(
+            errs => fromLeft(ResponseErrorInternal(JSON.stringify(errs))),
+            responseType =>
+              responseType.status !== 200
+                ? fromLeft(toErrorServerResponse(responseType))
+                : taskEither.of(responseType.value)
+          )
+      )
+  );
+
+/**
+ * Handles requests for updating a service by given serviceId and a Service Payload.
+ */
+export function GetUpdateServiceHandler(
+  apiClient: ReturnType<APIClient>
+): IGetCreateServiceHandler {
+  return (_, apiAuth, ___, ____, serviceId, servicePayload) => {
+    return updateServiceTask(
+      apiClient,
+      servicePayload,
+      serviceId,
+      apiAuth.subscriptionId
+    )
+      .chain(service =>
+        getSubscriptionKeysTask(apiClient, serviceId).map(subscriptionKeys =>
+          ResponseSuccessJson({
+            ...service,
+            ...subscriptionKeys
+          })
+        )
+      )
+      .fold<ResponseTypes>(identity, identity)
+      .run();
+  };
+}
+
+/**
+ * Wraps a GetUpdateService handler inside an Express request handler.
+ */
+export function GetUpdateService(
```

Comment:
```suggestion
export function UpdateService(
```

Reply:
Fixed in [98db5c7](https://github.com/pagopa/io-functions-services/pull/68/commits/98db5c77eefe2f2bbce99559596e00feed83aa0c)